src/sqlfluff/rules/L003.py

```diff
@@ -56,241 +221,165 @@ def _make_indent(
         num: int = 1, tab_space_size: int = 4, indent_unit: str = "space"
     ) -> str:
         if indent_unit == "tab":
-            base_unit = "\t"
-        elif indent_unit == "space":
-            base_unit = " " * tab_space_size
-        else:
-            raise ValueError(
-                f"Parameter indent_unit has unexpected value: `{indent_unit}`. Expected"
-                " `tab` or `space`."
-            )
-        return base_unit * num
+            return "\t" * num
+        if indent_unit == "space":
+            return " " * tab_space_size * num
+
+        raise ValueError(
+            f"Parameter indent_unit has unexpected value: `{indent_unit}`. Expected"
+            " `tab` or `space`."
+        )
 
     @staticmethod
     def _segment_length(elem: BaseSegment, tab_space_size: int):
-        # Start by assuming the typical case, where we need not consider slices
-        # or templating.
-        raw = elem.raw
-
-        # If it's whitespace, it might be a mixture of literal and templated
-        # whitespace. Check for this.
-        if elem.is_type("whitespace") and elem.is_templated:
-            # Templated case: Find the leading *literal* whitespace.
-            assert elem.pos_marker
-            templated_file = elem.pos_marker.templated_file
-            # Extract the leading literal whitespace, slice by slice.
-            raw = ""
-            for raw_slice in Segments(
-                elem, templated_file=templated_file
-            ).raw_slices.select(loop_while=rsp.is_slice_type("literal")):
-                # Compute and append raw_slice's contribution.
-                raw += sp.raw_slice(elem, raw_slice)
-
-        # convert to spaces for convenience (and hanging indents)
-        return raw.replace("\t", " " * tab_space_size)
+        return _segment_length(elem, tab_space_size)
```

Comment:
This seems odd -- a one-lined function that forwards to another function. Should we consider eliminating this?

Reply:
Just didn't want to change the test suite on this pass.
Will do a pass that removes this sort of thing - and makes a choice between functions, class methods, static methods and metods (we use a random mix at the moment)